cdap-watchdog/src/main/java/co/cask/cdap/logging/plugins/LocationManager.java

```diff
@@ -175,9 +181,36 @@ public void close() {
    */
   @Override
   public void flush() throws IOException {
+    Iterator<Map.Entry<LocationIdentifier, LocationOutputStream>> iter = activeLocations.entrySet().iterator();
+
+    while (iter.hasNext()) {
+      Map.Entry<LocationIdentifier, LocationOutputStream> entry = iter.next();
+      entry.getValue().flush();
+
+      if (shouldCloseFile(entry)) {
+        entry.getValue().close();
+        iter.remove();
```

Comment:
remove before closing, so if there is an exception, we wont be having a reference to output stream that can be in a bad state.